mindmaps-graql/src/main/java/io/mindmaps/graql/internal/reasoner/predicate/ValuePredicate.java

```diff
@@ -83,13 +85,15 @@ public int hashCode() {
     public boolean isEquivalent(Object obj){
         if (!(obj instanceof ValuePredicate)) return false;
         ValuePredicate a2 = (ValuePredicate) obj;
-        return this.getVal().equals(a2.getVal());
+        return this.predicate.getClass().equals(a2.predicate.getClass()) &&
```

Comment:
at the moment it is hence two predicates can have the same value and be different (differing by operators <, >, !=, =) and that test checks that as different operators are handled by different classes